src/client/Microsoft.Identity.Client/Platforms/netstandard13/NetStandard13CryptographyManager.cs

```diff
@@ -65,10 +69,18 @@ public byte[] Decrypt(byte[] encryptedMessage)
         /// <inheritdoc />
         public byte[] SignWithCertificate(string message, X509Certificate2 certificate)
         {
-            using (var key = certificate.GetRSAPrivateKey())
+            if (!s_certificateToRsaMap.TryGetValue(certificate.Thumbprint, out RSA rsa))
             {
-                return key.SignData(Encoding.UTF8.GetBytes(message), HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+                if (s_certificateToRsaMap.Count >= s_maximumMapSize)
```

Comment:
It seems like there's a lot of code duplication between net461, netcore and netstandard. Perhaps move it in a single location.

Reply:
@bgavrilMS I moved the code into CryptographyManager class. Does that look appropriate?

Reply:
LGTM